fuse_ll/src/fuse/mount.rs

```diff
@@ -543,54 +564,25 @@ pub fn mount(mount_point: &Path, options: &[&str]) -> RawFd {
 
     let mntpath = CString::new(cstr_path).expect("CString::new failed");
     let fstype = CString::new("osxfuse").expect("CString::new failed");
-    let fsname = CString::new("macfuse").expect("CString::new failed");
-    let fstypename = CString::new("").expect("CString::new failed");
-    let volname = CString::new("OSXFUSE Volume 0 (macfuse)").expect("CString::new failed");
-
-    // (fuse_mount_args) args = {
-    //     mntpath = "/private/tmp/hello"
-    //     fsname = "macfuse@osxfuse0"
-    //     fstypename = ""
-    //     volname = "OSXFUSE Volume 0 (macfuse)"
-    //     altflags = 64
-    //     blocksize = 4096
-    //     daemon_timeout = 60
-    //     fsid = 0
-    //     fssubtype = 0
-    //     iosize = 65536
-    //     random = 1477356727
-    //     rdev = 587202560
-    //   }
+
     let mut mntpath_slice = [0u8; MAXPATHLEN];
     copy_slice(mntpath.as_bytes(), &mut mntpath_slice);
-    let mut fsname_slice = [0u8; MAXPATHLEN];
-    copy_slice(fsname.as_bytes(), &mut fsname_slice);
-    let mut fstypename_slice = [0u8; MFSTYPENAMELEN];
-    copy_slice(fstypename.as_bytes(), &mut fstypename_slice);
-    let mut volname_slice = [0u8; MAXPATHLEN];
-    copy_slice(volname.as_bytes(), &mut volname_slice);
-
-    let mut mnt_args = FuseMountArgs {
-        mntpath: mntpath_slice,
-        fsname: fsname_slice,
-        fstypename: fstypename_slice,
-        volname: volname_slice,
-        altflags: FUSE_MOPT_DEBUG | FUSE_MOPT_FSNAME | FUSE_MOPT_NO_APPLEXATTR,
-        blocksize: FUSE_DEFAULT_BLOCKSIZE,
-        daemon_timeout: FUSE_DEFAULT_DAEMON_TIMEOUT,
-        fsid: 0,
-        fssubtype: FUSE_FSSUBTYPE_UNKNOWN,
-        iosize: FUSE_DEFAULT_IOSIZE,
-        random: drandom,
-        rdev: sb.st_rdev as u32,
-    };
+
+    args.set_mntpath(mntpath_slice);
+    args.set_random(drandom);
+    args.set_rdev(sb.st_rdev as u32);
```

Comment:
as conversion will trigger clippy lint warning, consider implement From trait for conversion

Reply:
I will make another PR to implement the From trait, since this PR is only supposed to use real mount options for macos.